QUARKUS-5477.md

```diff
@@ -0,0 +1,36 @@
+# QUARKUS-5477 Add OIDC Client SPI
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-5477
+
+Add OIDC Client SPI
+
+Upstream PRs:
+- https://github.com/quarkusio/quarkus/pull/42879
+
+Upsteram issue:
+- https://github.com/quarkusio/quarkus/issues/42878
+
+Documentation:
+- https://quarkus.io/guides/security-openid-connect-client-reference#oidc-client-spi
+
+## Scope of testing
+The `quarkus-oidc-client-spi` needs either the `quarkus-oidc-client` or `quarkus-oidc-client-deployment` dependency to be present.
+The implemented tests will validate the `TokenProvider` ability to generate access tokens for the default OIDC client.
+This implementation excludes access token provision for custom OIDC clients.
+
```

Comment:
This is optional, but would it be helpful to include the test cases you plan to implement?

Reply:
> This is optional, but would it be helpful to include the test cases you plan to implement?

I describing them in next section and don't want to duplicate things if not necessary